### PR TITLE
Explore: Cleanup redundant state variables and unused actions

### PR DESCRIPTION
**Pull request by aocenas**
- Removed state variables which could be easily derived from existing state.
- Removed unused actions and their reducers.
- Add some comments.

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
👍Added comments are super useful, awesome that you have added them 
👍Great job on removing unused reducers and actions
👍Great job with refactors for explore start page and explore datasources 

LGTM 🚀

File: public/app/features/explore/ExploreToolbar.tsx [hunk 1]
```diff
@@ -344,21 +347,16 @@ const mapStateToProps = (state: StoreState, { exploreId }: OwnProps): StateProps
...
   } = exploreItem;
-  const selectedDatasource = datasourceInstance
-    ? exploreDatasources.find(datasource => datasource.name === datasourceInstance.name)
-    : undefined;
+
   const hasLiveOption =
-    datasourceInstance && datasourceInstance.meta && datasourceInstance.meta.streaming && mode === ExploreMode.Logs
-      ? true
-      : false;
+    datasourceInstance && datasourceInstance.meta && datasourceInstance.meta.streaming && mode === ExploreMode.Logs;
```
Comment (davkal) [inline public/app/features/explore/ExploreToolbar.tsx]: Could do with `foo?.bar` syntax here too.

**Review by davkal — APPROVED** [review]
Nice work!

**Review by kaydelaney — APPROVED** [review]
great PR! lgtm

**Pull request merged**